trunk/includes/pardot-plugin-class.php

```diff
@@ -907,12 +907,16 @@ static function convert_embed_code_https( $embed_code ) {
 			 */
 			$reg_exUrl = apply_filters("pardot_https_regex", "/(http|https)\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,63}(\/\S*)?/");
 			preg_match( $reg_exUrl, $embed_code, $url );
-			/**
-			 * Replace whatever is there with the approved Pardot HTTPS URL
-			 */
-			$urlpieces  = parse_url($url[0]);
-			$httpsurl   = 'https://go.pardot.com' . $urlpieces['path'];
-			$embed_code = preg_replace( $reg_exUrl, $httpsurl, $embed_code );
+			
+			// Check if default domain is already HTTPS
+			if ( !strcasecmp(substr( $url[0], 0, 8 ), "https://") ) {
```

Comment:
Would this affect the assets that previously not using HTTPS?

Reply:
Thanks for pointing out the bug!  I had my conditional logic backwards. 😉  The plan is to first check if the asset is already using HTTPS.  If it is, just go with the domain it is already using.  If it is using HTTP, then we replace the domain with https://go.pardot.com, so that it doesn't cause issues with their WordPress site.  Users are likely checking that checkbox because they want assets to use HTTPS to prevent the issue of using an HTTP iframe within a HTTPS site.  Browsers don't like that very much.  I am planning to change the label on the "Use HTTPS?" checkbox to be more accurate and say some along the lines of "Force HTTPS", because if an asset is already configured to use a HTTPS domain, it will do so, regardless if that checkbox is selected.

Reply:
Wait, now I'm confused. If the goal of the PR is to rewrite URLs for assets that are **not** using HTTPS, then the original draft looks correct. The new behavior appears to rewrite URLs for all HTTPS assets to https://go.pardot.com, which seems undesirable.

Reply:
**Existing Behavior**
If the "Use HTTPS?" checkbox is enabled, all domains will be overwritten to use `https://go.pardot.com`, regardless of whether it uses HTTPS already or not.

**New Behavior**
When the "Always Use HTTPS" checkbox is enabled (I changed the name), we first check if the domain is already using HTTPS.  If it already using HTTPS, we let it pass through unchanged.  If it is not using HTTPS, we change the domain to use `https://go.pardot.com`.  This ensure that HTTPS is always used when the checkbox is enabled.

Please note that the preview above is out of date.  I accidentally had the conditional logic reversed.

Reply:
wait, it seems to me if `if ( strcasecmp(substr( $url[0], 0, 8 ), "https://")` passed, then it will use go.pardot.com?
Also, could be nitpick, we changed the name to "Always Use HTTPS" is it more like encourage people to check that box? If that the case, do we want to just enforce people use https instead of giving them option?

Reply:
`strcasecmp` returns 0 if the strings are equivalent, which will equate the false and skip this conditional logic.  It is confusing, hence why I messed up this conditional initially.  So if the URL passed in uses HTTPS, it will not convert the domain to go.pardot.com.

The name change to "Always Use HTTPS" was more so done to improve the accuracy checkbox label.  If an asset already uses a HTTPS tracker domain, it will still be used in WordPress, regardless of that checkbox.  Thus the checkbox acts as an override.  Since we don't require HTTPS for Pardot tracker domains, I think we should still give them the option.  We also don't want to break things for existing customers.